sys/linux/sys.txt

```diff
@@ -359,6 +360,8 @@ sysfs$2(option const[2], fsindex intptr, fsname buffer[out])
 sysfs$3(option const[3])
 statfs(path ptr[in, filename], buf buffer[out])
 fstatfs(fd fd, buf buffer[out])
+fsinfo$path(dfd const[AT_FDCWD], path ptr[in, filename], params ptr[in, fsinfo_params], params_size bytesize[params], buf buffer[out], buffer_size bytesize[buf])
+fsinfo$fd(dfd fd, path ptr[in, filename], params ptr[in, fsinfo_params], params_size bytesize[params], buf buffer[out], buffer_size bytesize[buf])
```

Comment:
This way of splitting `fsinfo` into `fsinfo$path` and `fsinfo$fd` doesn't really make sense.  All syscalls that take both a `dfd` and a `path` specify a file by the combination of these two arguments as follows:

* path begins with `/` => absolute path, `dfd` is ignored
* `dfd == AT_FDCWD` => path relative to current working dir
* `path == ""` and `AT_EMPTY_PATH` is given => `dfd` itself
* else path relative to `dfd` directory

So to properly make an `$fd` version of one of these syscalls, it actually should hardcode an empty path and `AT_EMPTY_PATH`, and make `dfd` an `fd` rather than an `fd_dir`.

But if we did that for `fsinfo()`, we should also do that for all other `*at` syscalls like `statat()`, `linkat()`, `renameat()`, etc.

If would be great if you could figure out what is really the best approach to describe these syscalls.  Here's an example of a case where it seemed syzkaller wasn't able to generate a good reproducer due to the current descriptions: https://groups.google.com/g/syzkaller-bugs/c/TPsaVuP6lh8/m/GUtq5AovCAAJ

But for now, how about just going with `fsinfo()` with `dfd fd_dir, pathname ptr[in, filename]`?

Reply:
Thanks for the explanation! I modified fsinfo to be with dfd fd_dir, pathname ptr[in, filename] as suggested. I do have a few questions about this: when we describe a path as ptr[in, filename], is it relative or absolute? My guess is that it's absolute. If so, do we currently have a way to get relative filepath? As for creating the $fd version, I can imagine creating a $fd version for each of the fs related syscalls. What would be the problem with that? Too much repetitive works?

Reply:
> when we describe a path as ptr[in, filename], is it relative or absolute?

`docs/syscall_descriptions_syntax.md` doesn't say.  You'll have to check the code. 

> I can imagine creating a $fd version for each of the fs related syscalls. What would be the problem with that? Too much repetitive works?

Yes, it would be repetitive.  It could be worth it though.

Alternatively, maybe syzkaller just isn't trying empty filenames frequently enough.